include/NovelRT/Timing/Timestamp.h

```diff
@@ -0,0 +1,33 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_TIMING_DELTATIMESTAMP_H
+#define NOVELRT_TIMING_DELTATIMESTAMP_H
+
+#ifndef NOVELRT_H
+#error Please do not include this directly. Use the centralised header (NovelRT.h) instead!
+#endif
+
+namespace NovelRT::Timing {
+  class Timestamp {
+  private:
+    uint64_t _ticks;
+
+  public:
+    Timestamp(uint64_t ticks) : _ticks(ticks) {}
+
+    inline uint64_t getTicks() const {
+      return _ticks;
+    }
+
+    inline double getSecondsDouble() const {
+      return static_cast<double>(_ticks) / TicksPerSecond;
+    }
+
+    inline float getSecondsFloat() const {
+      return static_cast<float>(_ticks) / TicksPerSecond;
```

Comment:
This is going to need to be `static_cast<float>(getSecondsDouble())`. The former should likely also be called just `getSeconds()`.

It may also be good to assert the `float` result is not zero if the `double` result is not zero.

Reply:
For reference... 

The issue here is that `float` can only represent integers up to 2^24 accurately which is only `1.6777216 seconds` if you are using 100ns ticks.
After that, you start getting more and more off in the computed result.

`double` on the other hand can represent integers up to 2^53, which is `10424.999137431703 days`. Once you scale those down by `TicksPerSecond` and then convert to float, they will generally be within a range where the minor precision loss doesn't matter.